cmd/template/cluster/provider/azure.go

```diff
@@ -62,10 +66,30 @@ func WriteCAPZTemplate(out io.Writer, config ClusterCRsConfig) error {
 		VMSize:            "Standard_D4s_v3",
 	}
 
-	t := template.Must(template.New(config.FileName).Parse(azure.GetTemplate()))
-	err = t.Execute(out, data)
-	if err != nil {
-		return microerror.Mask(err)
+	for _, t := range azure.GetTemplates() {
```

Comment:
The basic idea here is: Send each template individually to the management API to get defaulting from webhooks. Use `dry-run` to just get the defaulted objects.
I use the dynamic client here to avoid a versioning mess with go modules.

Reply:
Wasn't there a problem with ordering of object creation and dry run because some objects require that another object already exists?

Reply:
The order is the same as returned by `GetTemplates()` - so they are ordered correctly here.